src/Sarif/Schemata/sarif-schema.json

```diff
@@ -254,63 +254,99 @@
       }
     },
 
-    "externalFiles": {
-      "description": "References to external files that should be inlined with the content of a root log file.",
+    "externalPropertyFile": {
+      "type": "object",
+      "properties": {
+        "fileLocation": {
+          "description": "The location of the external property file.",
+          "$ref": "#/definitions/fileLocation"
+        },
+
+        "instanceGuid": {
+          "description": "A stable, unique identifer for the external property file in the form of a GUID.",
+          "type": "string"
+        },
+
+        "itemCount": {
+          "description": "A non-negative integer specifying the number of items contained in the external property file.",
+          "type": "integer",
+          "minimum": 0
+        },
+
+        "properties": {
+          "description": "Key/value pairs that provide additional information about the external property file.",
+          "$ref": "#/definitions/propertyBag"
+        }
+      }
+    },
+
+    "externalPropertyFiles": {
+      "description": "References to external property files that should be inlined with the content of a root log file.",
       "additionalProperties": false,
       "type": "object",
       "properties": {
 
         "conversion": {
-          "description": "The location of a file containing a run.conversion object to be merged with the root log file.",
-          "$ref": "#/definitions/fileLocation"
+          "description": "An external property file containing a run.conversion object to be merged with the root log file.",
+          "$ref": "#/definitions/externalPropertyFile"
         },
 
-        "files": {
-          "description": "The location of a file containing a run.files object to be merged with the root log file.",
-          "$ref": "#/definitions/fileLocation"
+        "graphs": {
+          "description": "An external property file containing a run.graphs object to be merged with the root log file.",
+          "$ref": "#/definitions/externalPropertyFile"
         },
 
-        "graphs": {
-          "description": "The location of a file containing a run.graphs object to be merged with the root log file.",
-          "$ref": "#/definitions/fileLocation"
+        "resources": {
+          "description": "An external property file containing a run.resources object to be merged with the root log file.",
+          "$ref": "#/definitions/externalPropertyFile"
         },
 
-        "invocations": {
-          "description": "An array of locations of files containing arrays of run.invocation objects to be merged with the root log file.",
+        "properties": {
+          "description": "An external property file containing a run.properties object to be merged with the root log file.",
+          "$ref": "#/definitions/externalPropertyFile"
```

Comment:
>externalPropertyFile [](start = 33, length = 20)

@lgolding, note that our spec actually says the "properties" value is an external property file. this item doesn't actually have a property bag. i assume that users can populate run.properties if they need anything specific around this container.

Reply:
I am absolutely fine with the `externalPropertyFiles` object not itself having a property bag. I think Jim objected to that but I am ok with it, and I agree that `run.properties` will service.

But see my other comment about this issue, in the visitor code.

---
In reply to: [244831292](https://github.com/Microsoft/sarif-sdk/pull/1194#discussion_r244831292) [](ancestors = 244831292)